internal/declarative/v2/reconciler.go

```diff
@@ -332,7 +346,9 @@ func (r *Reconciler) checkTargetReadiness(
 	crStateInfo, err := resourceReadyCheck.Run(ctx, clnt, manifest, target)
 	if err != nil {
 		r.Event(manifest, "Warning", "ResourceReadyCheck", err.Error())
-		manifest.SetStatus(status.WithState(shared.StateError).WithErr(err))
+		if manifest.GetDeletionTimestamp().IsZero() {
+			manifest.SetStatus(status.WithState(shared.StateError).WithErr(err))
```

Comment:
do we really need this condition? What I mean is, if this `resourceReadyCheck.Run` return error, it always put `crStateInfo.State` in `shared.StateError`, no matter manifest cr is under delete or not.

And if manifest CR is under delete, it may happen module CR is already deleted during ready check, but for this case, you already have this condition to make sure not return error if is not found.
```
	err := clnt.Get(ctx, client.ObjectKeyFromObject(moduleCR), moduleCR)
	if err != nil {
		if util.IsNotFound(err) && !manifest.DeletionTimestamp.IsZero() {
			return declarative.StateInfo{State: shared.StateDeleting}, nil
		}
		return declarative.StateInfo{State: shared.StateError}, fmt.Errorf("failed to fetch resource: %w", err)
	}
```

Reply:
In theory you are correct, I will give it a shot and remove if everything works as expected

Reply:
Removed it